src/libraries/Microsoft.PowerFx.Interpreter/PatchFunction.cs

```diff
@@ -0,0 +1,204 @@
+using Microsoft.PowerFx.Core.Utils;
+using Microsoft.PowerFx.Syntax;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Core.Localization.TexlStrings;
+
+namespace Microsoft.PowerFx.Interpreter
+{
+    /*
+         * @@@JYL@@@
+         * Verify before pushing it.
+         * => verify that Patch() fails in a non-behavior context.
+         * => Type checking - what if we pass the wrong type to patch (too few fields, too many fields, not a record, etc).
+         * => PowerApps lets pass multiple records - this is determined by CheckInvocation, which we copied from PA.
+         * => Will need to confirm corner case behavior, ie, what if we pass errors, etc. (We can probably do that in a followup PR, just to unblock usage now)
+         * => Trying to patch on an Immutable object. We should get an error.
+         */
+
+    internal abstract class PatchAndValidateRecordFunctionBase : BuiltinFunction
+    {
+        public PatchAndValidateRecordFunctionBase(DPath theNamespace, string name, TexlStrings.StringGetter description, FunctionCategories fc, DType returnType, BigInteger maskLambdas, int arityMin, int arityMax, params DType[] paramTypes)
+           : base(theNamespace, name, /*localeSpecificName*/string.Empty, description, fc, returnType, maskLambdas, arityMin, arityMax, paramTypes)
+        {
+        }
+
+        public PatchAndValidateRecordFunctionBase(string name, TexlStrings.StringGetter description, FunctionCategories fc, DType returnType, BigInteger maskLambdas, int arityMin, int arityMax, params DType[] paramTypes)
+            : this(DPath.Root, name, description, fc, returnType, maskLambdas, arityMin, arityMax, paramTypes)
+        {
+        }
+
+        public override bool CheckInvocation(TexlBinding binding, TexlNode[] args, DType[] argTypes, IErrorContainer errors, out DType returnType, out Dictionary<TexlNode, DType> nodeToCoercedTypeMap)
+        {
+            Contracts.AssertValue(binding);
+            Contracts.AssertValue(args);
+            Contracts.AssertAllValues(args);
+            Contracts.AssertValue(argTypes);
+            Contracts.Assert(args.Length == argTypes.Length);
+            Contracts.AssertValue(errors);
+
+            var isValid = CheckInvocation(args, argTypes, errors, out returnType, out nodeToCoercedTypeMap);
+
+            return isValid;
+        }
+    }
+
+    internal class PatchRecordFunction : PatchAndValidateRecordFunctionBase, IAsyncTexlFunction
+    {
+        public override bool IsSelfContained => false;
+
+        public PatchRecordFunction()
+            : base("Patch", TexlStrings.AboutPatch, FunctionCategories.Table | FunctionCategories.Behavior, DType.EmptyRecord, 0, 2, int.MaxValue, DType.EmptyRecord, DType.EmptyRecord)
+        {
+        }
+
+        public override IEnumerable<TexlStrings.StringGetter[]> GetSignatures()
+        {
+            yield return new[] { TexlStrings.AboutPatch, TexlStrings.AboutPatch, TexlStrings.AboutPatch };
```

Comment:
>AboutPatch [](http://example.com/codeflow?start=45&length=10)

This should be the arguments, not the about for the function

Reply:
Resolved to both PatchRecordFunction and PatchFunction functions. Let me know a way to test these locally is any.